pkg/endpoint/endpoint.go

```diff
@@ -104,10 +104,21 @@ func Listen(ctx context.Context, config Config) (ETCDConfig, error) {
 	endpoint := endpointURL(config, listener)
 	logrus.Infof("Kine available at %s", endpoint)
 
+	// Return a empty tls config if we do not receive any tls config
+	if config.ServerTLSConfig == (Config{}.ServerTLSConfig) {
+		return ETCDConfig{
+			LeaderElect: leaderelect,
+			Endpoints:   []string{endpoint},
+			TLSConfig:   tls.Config{},
+		}, nil
+	}
+
 	return ETCDConfig{
 		LeaderElect: leaderelect,
 		Endpoints:   []string{endpoint},
-		TLSConfig:   tls.Config{},
+		TLSConfig: tls.Config{
```

Comment:
This bit should be all you need - since the `ServerTLSConfig` field is not a pointer, it cannot be nil. If `config.ServerTLSConfig.CAFile` is an empty string, then it's the same as not setting it at all - there's no point in having a separate block for it.